go/libraries/doltcore/servercfg/yaml_config.go

```diff
@@ -127,21 +127,21 @@ type YAMLConfig struct {
 	LogLevelStr       *string                `yaml:"log_level,omitempty"`
 	MaxQueryLenInLogs *int                   `yaml:"max_logged_query_len,omitempty"`
 	EncodeLoggedQuery *bool                  `yaml:"encode_logged_query,omitempty"`
-	BehaviorConfig    BehaviorYAMLConfig     `yaml:"behavior"`
-	UserConfig        UserYAMLConfig         `yaml:"user"`
-	ListenerConfig    ListenerYAMLConfig     `yaml:"listener"`
+	BehaviorConfig    BehaviorYAMLConfig     `yaml:"behavior,omitempty"`
+	UserConfig        UserYAMLConfig         `yaml:"user,omitempty"`
+	ListenerConfig    ListenerYAMLConfig     `yaml:"listener,omitempty"`
 	PerformanceConfig *PerformanceYAMLConfig `yaml:"performance,omitempty"`
 	DataDirStr        *string                `yaml:"data_dir,omitempty"`
 	CfgDirStr         *string                `yaml:"cfg_dir,omitempty"`
-	MetricsConfig     MetricsYAMLConfig      `yaml:"metrics"`
-	RemotesapiConfig  RemotesapiYAMLConfig   `yaml:"remotesapi"`
+	MetricsConfig     MetricsYAMLConfig      `yaml:"metrics,omitempty"`
+	RemotesapiConfig  RemotesapiYAMLConfig   `yaml:"remotesapi,omitempty"`
```

Comment:
We want ClusterConfig and MetricsConfig and Jwks the bottom of the file, since they are only for advanced users. Since the yaml marshaler outputs in struct field order, that means we should move these fields to the end of the struct.